compiler/src/iree/compiler/Codegen/ROCDL/Passes.cpp

```diff
@@ -0,0 +1,184 @@
+// Copyright 2024 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "iree-dialects/Dialect/LinalgExt/Passes/Passes.h"
+#include "iree/compiler/Codegen/Common/GPU/Passes.h"
+#include "iree/compiler/Codegen/Common/Passes.h"
+#include "iree/compiler/Codegen/LLVMGPU/Passes.h"
+#include "iree/compiler/Codegen/ROCDL/Passes.h"
+#include "iree/compiler/Dialect/Util/Transforms/Passes.h"
+#include "mlir/Conversion/AffineToStandard/AffineToStandard.h"
+#include "mlir/Conversion/ComplexToStandard/ComplexToStandard.h"
+#include "mlir/Conversion/SCFToControlFlow/SCFToControlFlow.h"
+#include "mlir/Conversion/VectorToGPU/VectorToGPU.h"
+#include "mlir/Dialect/Arith/Transforms/Passes.h"
+#include "mlir/Dialect/Func/IR/FuncOps.h"
+#include "mlir/Dialect/Func/Transforms/Passes.h"
+#include "mlir/Dialect/GPU/Transforms/Passes.h"
+#include "mlir/Dialect/Linalg/Passes.h"
+#include "mlir/Dialect/MemRef/Transforms/Passes.h"
+#include "mlir/Pass/PassManager.h"
+#include "mlir/Pass/PassOptions.h"
+#include "mlir/Pass/PassRegistry.h"
+#include "mlir/Transforms/Passes.h"
+
+#define DEBUG_TYPE "iree-rocdl-lowering-pass-pipeline"
+
+namespace mlir::iree_compiler {
+
+//===----------------------------------------------------------------------===//
+// Common Pass Recipes
+//===----------------------------------------------------------------------===//
+
+// Add passes to make the address computation more explicit and optimize them.
+//
+// The idea here is to be less dependent on what the LLVM backend is able to do,
+// by heavy lifting most of the work while we still have the information about
+// loops.
+//
+// Note that this needs to run before SCF -> CF.
+static void addLowerAndOptimizeAddressComputationPasses(OpPassManager &pm) {
```

Comment:
THere is a lot of duplication here. This is exactly what the CUDA side does? I am OK with some replication, but these seem like generic utilities.

Reply:
Changed to reuse existing code.